optimizer_wrapper.rb

```diff
@@ -221,14 +221,23 @@ def self.solve(service_vrp, job = nil, block = nil)
         end
       }
 
+      # vrp.periodic_heuristic check the first_solution_stategy which may change right after periodic heuristic
+      periodic_heuristic_flag = vrp.periodic_heuristic?
       # TODO: refactor with dedicated class
       if vrp.schedule?
         periodic = Interpreters::PeriodicVisits.new(vrp)
         vrp = periodic.expand(vrp, job, &block)
         optim_solution = vrp.configuration.preprocessing.heuristic_result if vrp.periodic_heuristic?
+        if vrp.configuration.resolution.solver && vrp.configuration.preprocessing.first_solution_strategy == ['periodic']
+          vrp.configuration.resolution.duration -= optim_solution.elapsed
+          if vrp.configuration.resolution.minimum_duration && vrp.configuration.resolution.duration
+            vrp.configuration.resolution.minimum_duration *= 1 - optim_solution.elapsed / vrp.configuration.resolution.duration
+          end
+          vrp.configuration.preprocessing.first_solution_strategy = ['global_cheapest_arc']
+        end
```

Comment:
I am not sure if we need this time correction logic because the heuristic is quite fast (especially for small instances)  but also we always argue that the min/max_durations are for "solver time" but this elapsed time includes all the overheads plus it might lead to negative time at this current state
```ruby
vrp.configuration.resolution.duration -= optim_solution.elapsed
if vrp.configuration.resolution.minimum_duration && vrp.configuration.resolution.duration
     vrp.configuration.resolution.minimum_duration *= 1 - optim_solution.elapsed / vrp.configuration.resolution.duration
end
```

Lets not override user's first_solution_strategy choices:
```suggestion
        if vrp.configuration.resolution.solver && vrp.periodic_heuristic?
          first_solution_strategy = vrp.configuration.preprocessing.first_solution_strategy
          first_solution_strategy -= 'periodic'
          first_solution_strategy += 'global_cheapest_arc' if first_solution_strategy.empty?
        end
```

Reply:
negative time happened already :) 
![Screenshot from 2022-04-21 11-59-46](https://user-images.githubusercontent.com/20163961/164430994-7b17f9a8-0046-4292-825c-c3e35a9bd661.png)